diff -ur meson-0.55.0.orig/mesonbuild/compilers/mixins/intel.py meson-0.55.0/mesonbuild/compilers/mixins/intel.py
--- meson-0.55.0.orig/mesonbuild/compilers/mixins/intel.py	2020-01-23 12:51:19.000000000 -0000
+++ meson-0.55.0/mesonbuild/compilers/mixins/intel.py	2020-08-13 12:21:22.276409643 -0000
@@ -24,6 +24,7 @@
 import typing as T
 
 from ... import mesonlib
+from ...envconfig import get_env_var
 from .gnu import GnuLikeCompiler
 from .visualstudio import VisualStudioLikeCompiler
 
@@ -125,6 +126,22 @@
     def get_optimization_args(self, optimization_level: str) -> T.List[str]:
         return self.OPTIM_ARGS[optimization_level]
 
+    def get_compiler_dirs(self, env: 'Environment', name: str) -> T.List[str]:
+        '''
+        Get dirs from the compiler, either `libraries:` or `programs:`
+        '''
+        dirs = super().get_compiler_dirs(env, name)
+        # GCC includes $LIBRARY_PATH directories but the Intel compiler
+        # does not.
+        if name == 'libraries':
+            library_path = get_env_var(
+                self.for_machine,
+                env.is_cross_build(),
+                'LIBRARY_PATH')
+            if library_path:
+                dirs.extend(self._split_fetch_real_dirs(library_path))
+        return dirs
+
 
 class IntelVisualStudioLikeCompiler(VisualStudioLikeCompiler):
 
